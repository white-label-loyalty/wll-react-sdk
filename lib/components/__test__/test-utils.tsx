import { render } from '@testing-library/react';
import React from 'react';
import { CoreApiUrl, WllSdkProvider } from '../../context/WllSdkContext';
import { defaultTheme } from '../../utils/styling';

const defaultConfig = {
  apiKey: 'test-api-key',
  userToken: 'test-user-token',
  coreApiUrl: CoreApiUrl.STAGING_EU,
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <WllSdkProvider
      config={defaultConfig}
      theme={defaultTheme}
      navigationConfig={{}}
    >
      {children}
    </WllSdkProvider>
  );
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
