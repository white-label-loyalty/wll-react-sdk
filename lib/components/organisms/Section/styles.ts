import { StyleSheet } from 'react-native';
import { MAX_WIDTH } from '../../../constants';
import { useWllSdk } from '../../../context/WllSdkContext';
import { useResponsive } from '../../../hooks/useResponsive';
import { useResponsiveValue } from '../../../utils/responsiveHelper';

export const useSectionStyles = () => {
  const { isDesktop, isTablet } = useResponsive();
  const { theme } = useWllSdk();
  return StyleSheet.create({
    section: {
      width: '100%',
      maxWidth: MAX_WIDTH,
      marginHorizontal: 'auto',
      marginBottom: useResponsiveValue(
        theme.sizes.xxxxxl,
        theme.sizes.xxxxl,
        isDesktop,
        isTablet
      ),
    },
  });
};
