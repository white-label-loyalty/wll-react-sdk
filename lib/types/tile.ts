export enum CtaAction {
  sameWindow = 'SAME_WINDOW',
  newWindow = 'NEW_WINDOW',
}

export enum BadgeTileType {
  latestEarned = 'LATEST_EARNED',
  specificBadge = 'SPECIFIC_BADGE',
}

export enum TierTileType {
  currentTier = 'CURRENT_TIER',
  nextTier = 'NEXT_TIER',
  specificTier = 'SPECIFIC_TIER',
}

export enum TileType {
  Banner = 'BANNER',
  Points = 'POINTS',
  Content = 'CONTENT',
  Reward = 'REWARD',
  Badge = 'BADGE',
  RewardCategory = 'REWARD_CATEGORY',
  Tier = 'TIER',
}

export enum TileHeight {
  Half = 'HALF',
  Full = 'FULL',
}
type Badge = {
  id: string | null;
  name: string | null;
  description: string | null;
  artworkUrl: string | null;
  createdAt: string | null;
  updatedAt: string | null;
};
export class RewardCategory {
  name?: string | null;
  priority?: number;
  type?: string | null;
  id?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  description?: string | null;
  metadata?: string | null;
  pictureUrl?: string | null;
  rewards?: Reward[];
  parent?: string | null;
}

export class Reward {
  id?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  name?: string | null;
  pictureUrl?: string | null;
  description?: string | null;
  value?: number | null;
  price?: number;
  priority?: number;
  discounts?: any[] | null;
  redemptionChannels?: string[] | null;
  venues?: any[] | null;
  purchasableForAudiences?: string[] | null;
  category?: string | null;
  terms?: string | null;
  availability?: Availability;
  purchasable?: boolean;
  tier?: string | null;
  summary?: string | null;
  redemptionMessage?: string | null;
  visibilityCriteria?: string | null;
  logoUrl?: string | null;
  type?: string | null;
  codeType?: string | null;
  code?: string | null;
  purchaseExpiration?: string | null;
  hideCode?: boolean;
  notificationConfig?: string | null;
}
type Availability = {
  start: string | null;
  end: string | null;
};
type Effectivity = {
  start: string | null;
  end: string | null;
};

export type TierType = {
  id: string | null;
  name: string | null;
  description: string | null;
  artworkUrl: string | null;
  priority: number;
  pointsRequirement: number;
  calculation: string | null;
  accumulationPeriod: string | null;
  effectivity: Effectivity;
  createdAt: string | null;
  updatedAt: string | null;
  earnedPoints: number;
  attained: boolean;
};

export class BannerTileConfig {
  imageUrl?: string | null;
  title?: string | null;
  description?: string | null;
  ctaText?: string | null;
  ctaLink?: string | null;
  ctaAction?: CtaAction;
}

export class PointsTileConfig {
  title?: string | null;
  multiplier?: number;
  prefix?: string | null;
  suffix?: string | null;
  imageUrl?: string | null;
  points?: number;
}

export class ContentTileConfig {
  title?: string | null;
  subtitle?: string | null;
  imageUrl?: string | null;
}

export class RewardTileConfig {
  reward?: Reward;
  showPrice?: boolean;
}

export class BadgeTileConfig {
  badgeTileType?: BadgeTileType;
  badgeId?: string | null;
  badge?: Badge;
}

export class RewardCategoryTileConfig {
  categoryId?: string | null;
  allowDecorationOverlay?: boolean;
  rewardCategory?: RewardCategory;
}

export class TierTileConfig {
  tierId?: string | null;
  tier?: TierType;
  type?: TierTileType;
}

export type Tile = {
  id: string | null;
  type: TileType;
  active: boolean;
  createdAt: string | null;
  updatedAt: string | null;
  visibilityCriteria: object | null;
  tileHeight: TileHeight;
  configuration: TileConfig;
};

const getConfigForTileType = (tileType: TileType) => {
  switch (tileType) {
    case TileType.Banner:
      return BannerTileConfig;
    case TileType.Points:
      return PointsTileConfig;
    case TileType.Content:
      return ContentTileConfig;
    case TileType.Reward:
      return RewardTileConfig;
    case TileType.Badge:
      return BadgeTileConfig;
    case TileType.RewardCategory:
      return RewardCategoryTileConfig;
    case TileType.Tier:
      return TierTileConfig;
    default:
      throw new Error(`Tile type ${tileType} is not supported`);
  }
};

export type TileConfig = InstanceType<
  ReturnType<typeof getConfigForTileType>
>;
