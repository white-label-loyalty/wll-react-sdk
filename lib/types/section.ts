import { Tile } from './tile';

export enum SectionType {
  Grid = 'GRID',
  Banner = 'BANNER',
}

export type TSection = {
  id: string;
  type: SectionType;
  createdAt: string;
  updatedAt: string;
  title?: string;
  description?: string;
  tiles: Tile[];
  pointsMultiplier?: number;
  pointsPrefix?: string | null;
  pointsSuffix?: string | null;
  priority: number;
  active: boolean;
};
