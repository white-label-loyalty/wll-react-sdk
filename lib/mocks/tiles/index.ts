import { Tile, TileType } from '../../types/tile';
import { createBadgeTileMock } from './badgeTile';
import { createBannerTileMock } from './bannerTile';
import { createContentTileMock } from './contentTile';
import { createPointsTileMock } from './pointsTile';
import { createRewardCategoryTileMock } from './rewardCategoryTile';
import { createRewardTileMock } from './rewardTile';
export { createContentTileMock } from './contentTile';

export const createTileMock = (type: TileType, config?: any): Tile => {
  switch (type) {
    case TileType.Content:
      return createContentTileMock(config);
    case TileType.Reward:
      return createRewardTileMock(config);
    case TileType.Banner:
      return createBannerTileMock(config);
    case TileType.Points:
      return createPointsTileMock(config);
    case TileType.Badge:
      return createBadgeTileMock(config);
    case TileType.RewardCategory:
      return createRewardCategoryTileMock(config);
    case TileType.Tier:
    // TODO: Implement other tile types as they are needed
    default:
      throw new Error(`Tile type ${type} not yet implemented`);
  }
};
