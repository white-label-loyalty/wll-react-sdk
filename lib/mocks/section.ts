import { SectionType, TSection } from '../types/section';
import { Tile, TileType } from '../types/tile';
import { createTileMock } from './tiles';

export type SectionMockConfig = Partial<TSection> & {
  tiles?: Tile[];
};

const defaultConfig: TSection = {
  details: [],
  defaultLocale: 'en',
  name: 'Elite Member Benefits',
  type: SectionType.Grid,
  active: true,
  id: 'a35399a7-580a-4493-afff-24ef4facf3f4',
  createdAt: '2024-11-19T15:50:37.701Z',
  updatedAt: '2024-12-06T11:18:16.842Z',
  priority: 2,
  tiles: [createTileMock(TileType.Content)],
  title: 'Elite Member Benefits',
  locale: 'en',
  pointsSuffix: 'miles',
};

export const createSectionMock = (config?: SectionMockConfig): TSection => {
  const { tiles, ...sectionConfig } = config || {};
  
  // If tiles are provided in config, use those. Otherwise use default tiles
  const mergedTiles = tiles || defaultConfig.tiles;

  return {
    ...defaultConfig,
    ...sectionConfig,
    tiles: mergedTiles,
  };
};

// Helper function to create a section with specific tile types
export const createSectionWithTiles = (tiles: Tile[]): TSection => {
  return createSectionMock({ tiles });
};
