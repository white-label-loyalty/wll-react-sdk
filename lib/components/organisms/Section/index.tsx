import React, { createContext, useContext, useEffect, useState } from 'react';
import { Text, View } from 'react-native';
import { useWllSdk } from '../../../context/WllSdkContext';
import { SectionType, TSection } from '../../../types/section';
import { commonStyles } from '../../../utils/styling';
import { Skeleton } from '../../atoms';
import { Carousel, Grid } from '../../molecules';
import { useSectionStyles } from './styles';

type SectionContextType = {
  sectionData: TSection;
};

type SectionProps = {
  section?: TSection;
  sectionId?: string;
};

/**
 * SectionContext provides the current section data to child components.
 */
export const SectionContext = createContext<SectionContextType | undefined>(
  undefined
);

/**
 * Custom hook to access the SectionContext.
 */
export const useSectionContext = (): SectionContextType => {
  const context = useContext(SectionContext);
  if (!context) {
    throw new Error('useSectionContext must be used within a SectionProvider');
  }
  return context;
};

/**
 * Custom hook to fetch section data.
 */
const useSectionData = (
  section?: TSection,
  sectionId?: string
): {
  sectionData: TSection | null;
  isLoading: boolean;
  error: string | null;
} => {
  const { getSectionByID } = useWllSdk();
  const [sectionData, setSectionData] = useState<TSection | null>(
    section || null
  );
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    if (section) {
      setSectionData(section);
    } else if (sectionId) {
      const fetchSection = async () => {
        setIsLoading(true);
        try {
          const response = await getSectionByID(sectionId);
          if (response.status === 'success' && response.data) {
            setSectionData(response.data);
          } else {
            setError(response.error || 'Failed to fetch section data.');
          }
        } catch (err) {
          setError(
            err instanceof Error ? err.message : 'Failed to fetch section data.'
          );
        } finally {
          setIsLoading(false);
        }
      };
      fetchSection();
    }
  }, [section, sectionId, getSectionByID]);

  return { sectionData, isLoading, error };
};

/**
 * Component to display an empty state with a message.
 */
const EmptyState = ({ message }: { message: string }): JSX.Element => (
  <View
    style={commonStyles.emptyContainer}
    accessible
    accessibilityLabel={`Empty state: ${message}`}
  >
    <Text
      accessibilityElementsHidden={true}
      importantForAccessibility="no-hide-descendants"
    >
      {message}
    </Text>
  </View>
);

/**
 * The Section component renders a section based on its type (e.g., Banner, Grid).
 */
const Section = ({ section, sectionId }: SectionProps): JSX.Element | null => {
  const styles = useSectionStyles();
  const { sectionData, isLoading, error } = useSectionData(section, sectionId);

  if (!section && !sectionId) {
    console.warn(
      'Section component requires either section or sectionId prop.'
    );
    return null;
  }

  const renderSectionContent = (): JSX.Element | null => {
    if (isLoading) {
      return (
        <Skeleton aria-label="Loading section content" numberOfSquares={4} />
      );
    }

    if (error || !sectionData) {
      return <EmptyState message={error || 'No section data available.'} />;
    }

    // Filter out inactive tiles before passing to child components
    const activeTiles = sectionData.tiles.filter((tile) => tile.active);
    const sectionWithActiveTiles = { ...sectionData, tiles: activeTiles };

    switch (sectionData.type) {
      case SectionType.Banner:
        return <Carousel section={sectionWithActiveTiles} />;
      case SectionType.Grid:
        return <Grid section={sectionWithActiveTiles} />;
      default:
        console.warn(`Unknown section type: ${sectionData.type}`);
        return <EmptyState message="Unknown section type." />;
    }
  };

  return sectionData ? (
    <SectionContext.Provider value={{ sectionData }}>
      <View
        style={styles.section}
        accessible
        role="region"
        aria-label={`Section: ${sectionData.title || 'Untitled section'}`}
        aria-description={sectionData.description || undefined}
      >
        {renderSectionContent()}
      </View>
    </SectionContext.Provider>
  ) : null;
};

export default Section;
