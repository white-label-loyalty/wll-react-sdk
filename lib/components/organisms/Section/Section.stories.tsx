import { Section, TierTile } from "@/components/organisms";
import { SectionType } from "@/types/section";
import { TierTileType, Tile, TileHeight, TileType } from "@/types/tile";
import { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { View } from "react-native";

export default {
  title: "components/organisms/Section",
  component: Section,
  args: {
    section: {
      id: "1",
    },
  },
  decorators: [
    (Story) => (
      <View
        style={{
          width: "100%",
          height: "100%",
          alignContent: "center",
          justifyContent: "center",
        }}
      >
        <Story />
      </View>
    ),
  ],
} as Meta;

const Template: StoryFn<typeof Section> = (args) => <Section {...args} />;

export const GridLayout = Template.bind({});
GridLayout.args = {
  section: {
    id: "1",
    type: SectionType.Grid,
    createdAt: new Date(),
    updatedAt: new Date(),
    tenantId: "tenant1",
    visibilityCriteria: "all",
    tiles: [
      {
        id: "3",
        type: TileType.Tier,
        createdAt: new Date(),
        updatedAt: new Date(),
        tenantId: "tenant1",
        visibilityCriteria: "all",
        tileHeight: TileHeight.Full,
        configuration: {
          tierTileType: TierTileType.currentTier,
          tierId: "1",
        },
      },
      {
        id: "1",
        type: TileType.Tier,
        createdAt: new Date(),
        updatedAt: new Date(),
        tenantId: "tenant1",
        visibilityCriteria: "all",
        tileHeight: TileHeight.Half,
        configuration: {
          tierTileType: TierTileType.currentTier,
          tierId: "2",
        },
      },
      {
        id: "2",
        type: TileType.Tier,
        createdAt: new Date(),
        updatedAt: new Date(),
        tenantId: "tenant1",
        visibilityCriteria: "all",
        tileHeight: TileHeight.Half,
        configuration: {
          tierTileType: TierTileType.currentTier,
          tierId: "3",
        },
      },

      {
        id: "4",
        type: TileType.Tier,
        createdAt: new Date(),
        updatedAt: new Date(),
        tenantId: "tenant1",
        visibilityCriteria: "all",
        tileHeight: TileHeight.Full,
        configuration: {
          tierTileType: TierTileType.currentTier,
          tierId: "4",
        },
      },
      {
        id: "5",
        type: TileType.Tier,
        createdAt: new Date(),
        updatedAt: new Date(),
        tenantId: "tenant1",
        visibilityCriteria: "all",
        tileHeight: TileHeight.Half,
        configuration: {
          tierTileType: TierTileType.currentTier,
          tierId: "5",
        },
      },
      {
        id: "6",
        type: TileType.Tier,
        createdAt: new Date(),
        updatedAt: new Date(),
        tenantId: "tenant1",
        visibilityCriteria: "all",
        tileHeight: TileHeight.Half,
        configuration: {
          tierTileType: TierTileType.currentTier,
          tierId: "5",
        },
      },
    ],
  },
  renderItem: ({ tile }: { tile: Tile }) => <TierTile {...tile} />,
};

// export const BannerLayout = Template.bind({});
// BannerLayout.args = {
//   ...GridLayout.args,
//   section: {
//     ...GridLayout.args.section,
//     type: "BANNER",
//   },
//   renderItem: ({ tile }: { tile: Tile }) => (
//       <TierTile tile={tile} />,
//   ),
// };
