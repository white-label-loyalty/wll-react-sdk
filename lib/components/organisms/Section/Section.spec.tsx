import React from 'react';
import { createSectionMock } from '../../../mocks/section';
import { createTileMock } from '../../../mocks/tiles';
import { TileHeight, TileType } from '../../../types/tile';
import { render } from '../../__test__/test-utils';
import Section from './index';

const contentTile = createTileMock(TileType.Content, {
  title: 'Welcome to Platinum Status',
  body: 'Enjoy premium benefits including priority check-in, lounge access, and complimentary upgrades. Your elite journey begins here.',
  artworkUrl: 'https://ucarecdn.com/b129c5c2-08fd-4962-83b1-5c35301148c1/',
  tileHeight: TileHeight.Full,
});

const sectionMock = createSectionMock({
  tiles: [contentTile],
});

describe('<Section/>', () => {
  it('matches snapshot', () => {
    const { container } = render(<Section section={sectionMock} />);
    expect(container).toMatchSnapshot();
  });
});
