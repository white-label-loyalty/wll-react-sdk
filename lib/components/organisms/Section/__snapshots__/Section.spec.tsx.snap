// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section/> matches snapshot 1`] = `
<div>
  <div
    aria-label="Section: Elite Member Benefits"
    class="css-view-175oi2r r-marginBottom-1c6w074 r-marginInline-f8sm7e r-maxWidth-q90dru r-width-13qz1uu"
  >
    <div
      class="css-view-175oi2r"
    >
      <div
        class="css-view-175oi2r r-marginBottom-1ifxtd0"
      >
        <div
          class="css-text-146c3p1 r-fontSize-1ui5ee8 r-marginBottom-5oul0u"
          dir="auto"
          style="font-weight: 700; color: rgb(0, 0, 0);"
        >
          Elite Member Benefits
        </div>
      </div>
      <div
        class="css-view-175oi2r r-flexDirection-18u37iz r-flexWrap-1w6e6rj r-justifyContent-1wtj0ep r-width-13qz1uu"
      >
        <div
          class="css-view-175oi2r"
          style="width: calc(50% - 8px); margin-bottom: 16px; height: auto;"
        >
          <div
            class="css-view-175oi2r r-flex-13awgt0 r-flexDirection-eqz5dr r-height-1pi2tsx r-width-13qz1uu"
            data-testid="tile-container"
          >
            <div
              class="css-view-175oi2r r-flexBasis-1fq43b1 r-flexGrow-16y2uox r-flexShrink-1wbh5a2 r-width-13qz1uu"
            >
              <button
                aria-disabled="true"
                aria-label="Welcome to Platinum Status"
                class="css-view-175oi2r r-pointerEvents-12vffkv r-aspectRatio-1ujkv8a r-borderRadius-y47klf r-height-1pi2tsx r-overflow-1udh08x r-position-bnwqim r-width-13qz1uu"
                disabled=""
                role="button"
                style="background-color: rgb(248, 247, 252); opacity: 1; flex-direction: column; justify-content: flex-start; align-items: stretch;"
                tabindex="-1"
                type="button"
              >
                <button
                  aria-disabled="true"
                  aria-label="Welcome to Platinum Status"
                  class="css-view-175oi2r r-pointerEvents-12vffkv r-aspectRatio-1ujkv8a r-borderRadius-y47klf r-height-1pi2tsx r-overflow-1udh08x r-position-bnwqim r-width-13qz1uu"
                  disabled=""
                  role="button"
                  style="background-color: rgb(248, 247, 252); opacity: 1; flex-direction: column; justify-content: flex-start; align-items: stretch;"
                  tabindex="-1"
                  type="button"
                >
                  <div
                    class="css-view-175oi2r r-alignItems-1awozwy r-height-1pi2tsx r-justifyContent-1777fci r-position-bnwqim r-flexBasis-1t2qqvi r-objectFit-1t7uo4s r-width-13qz1uu"
                    style="background-color: rgba(57, 46, 215, 0.2);"
                  >
                    <div
                      class="css-view-175oi2r r-bottom-1p0dtai r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af"
                      style="background-color: rgba(57, 46, 215, 0.2); opacity: 1;"
                    />
                    <div
                      aria-label="Content image for Welcome to Platinum Status"
                      class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010 r-bottom-1p0dtai r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af"
                      data-testid="tile-media"
                      role="img"
                      style="opacity: 0;"
                    >
                      <div
                        class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
                        style="background-image: url(https://ucarecdn.com/b129c5c2-08fd-4962-83b1-5c35301148c1/);"
                      />
                      <img
                        alt="Content image for Welcome to Platinum Status"
                        class="css-accessibilityImage-9pa8cd"
                        draggable="false"
                        src="https://ucarecdn.com/b129c5c2-08fd-4962-83b1-5c35301148c1/"
                      />
                    </div>
                  </div>
                  <div
                    class="css-view-175oi2r r-display-6koalj"
                    data-testid="tile-content"
                    role="presentation"
                    style="justify-content: center;"
                  >
                    <div
                      class="css-view-175oi2r r-flex-13awgt0 r-justifyContent-1777fci r-paddingInline-3o4zer"
                    >
                      <h1
                        class="css-view-175oi2r"
                        data-testid="tile-header"
                        role="heading"
                        style="margin-bottom: 8px; margin-top: 12px; flex-direction: row; align-items: center; justify-content: space-between;"
                      >
                        <h1
                          aria-label="Welcome to Platinum Status"
                          class="css-text-146c3p1 r-maxWidth-dnmrzs r-overflow-1udh08x r-textOverflow-1udbk01 r-whiteSpace-3s2u2q r-wordWrap-1iln25a"
                          data-testid="tile-title"
                          dir="auto"
                          role="heading"
                          style="color: rgb(0, 0, 0); font-size: 24px; font-weight: bold;"
                        >
                          Welcome to Platinum Status
                        </h1>
                      </h1>
                      <div
                        aria-label="Enjoy premium benefits including priority check-in, lounge access, and complimentary upgrades. Your elite journey begins here."
                        class="css-text-146c3p1 r-WebkitBoxOrient-8akbws r-display-krxsd3 r-maxWidth-dnmrzs r-overflow-1udh08x r-textOverflow-1udbk01"
                        data-testid="tile-body"
                        dir="auto"
                        style="color: rgb(51, 51, 51); font-size: 14px;"
                      >
                        Enjoy premium benefits including priority check-in, lounge access, and complimentary upgrades. Your elite journey begins here.
                      </div>
                    </div>
                  </div>
                </button>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
